sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java

```diff
@@ -132,6 +132,29 @@ public static boolean isOwnedByThisFramework(Resource resource, FrameworkInfo fr
         return hasResourceId && matchingRoles;
     }
 
+    public static Collection<Resource.ReservationInfo> getReservations(Resource resource) {
+        Set<Resource.ReservationInfo> reservations =
+                new HashSet<>(resource.getReservationsList().stream().collect(Collectors.toSet()));
+        if (resource.hasReservation()) {
+            reservations.add(resource.getReservation());
+        }
+
+        return reservations;
+    }
+
+    public static Collection<Resource.ReservationInfo> getDynamicReservations(Resource resource) {
+        return getReservations(resource).stream()
+                .filter(reservationInfo -> reservationInfo.hasType())
+                .filter(reservationInfo -> reservationInfo.getType()
+                        .equals(Protos.Resource.ReservationInfo.Type.DYNAMIC))
+                .collect(Collectors.toSet());
+    }
+
+    public static boolean isProcessable(Protos.Resource resource, Protos.FrameworkInfo frameworkInfo) {
+        boolean hasDynamicReservations = ResourceUtils.getDynamicReservations(resource).size() > 0;
```

Comment:
If you wanted to optimize out pulling the dynamic reservations, you could check isOwnedBy first, return if true, then check dynamic reservations.

Reply:
They're of a similar time complexity.